### PR TITLE
Cleanup old and unused models

### PR DESCRIPTION
**Pull request by paranoicsan**
These changes should clean up old and unused models in the project.

I tried to split the changes per commit basis to simplify the review. I will squash them all right before merging.

### CODE REVIEW

File: app/models/settings.rb [hunk 1]
```diff
@@ -2,16 +2,26 @@
 
 class Settings < ActiveRecord::Base
   class << self
-    def editing_enabled?
-      settings.editing_enabled?
+    def method_missing(name, *args, &block)
```
Comment (andersoncardoso) [inline app/models/settings.rb]: I don't like overriding method_missing. Perhaps on this case we can just define the 2 methods we want:
```
def self.[](key)
   settings.data[key.to_s]
end

def self.[]=(key, value)
  settings.update data: settings.data.merge k.to_so => value
end
```
we probably can also remove the `respond_to_missing?` in this case.

File: db/migrate/20180307172827_remove_resource_backup.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class RemoveResourceBackup < ActiveRecord::Migration
+  def change
+    drop_table :resource_backups
```
Comment (andersoncardoso) [inline db/migrate/20180307172827_remove_resource_backup.rb]: would be good to left the block definition for the rollback

File: db/migrate/20180307173100_remove_resource_requirement.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class RemoveResourceRequirement < ActiveRecord::Migration
+  def change
+    drop_table :resource_requirements
```
Comment (andersoncardoso) [inline db/migrate/20180307173100_remove_resource_requirement.rb]: here too, is missing a rollback definition

File: db/migrate/20180307172827_remove_resource_backup.rb (hunk 2 above)
Comment (paranoicsan) [reply db/migrate/20180307172827_remove_resource_backup.rb]: Yeah, the same as we had on UB core a few days ago 😄

File: app/models/settings.rb (hunk 1 above)
Comment (paranoicsan) [reply app/models/settings.rb]: I think we can

**Pull request merged**